File: 830. Positions of Large Groups.java
class Solution {
    public static List<List<Integer>> largeGroupPositions(String S) {
        if (S == null || S.length() < 3) {
            return new ArrayList<>();
        }
        int len = S.length();
        char[] chs = S.toCharArray();
        List<List<Integer>> res = new ArrayList<>();
        for (int i = 0; i < len; i++) {
            int end = i;
            while (end < len - 1 && chs[end] == chs[end + 1]) {
                end++;
            }
            if (end - i + 1 >= 3) {
                res.add(Arrays.asList(i, end));
            }
            i = end;
        }
        return res;
    }
}

File: Go/830. Positions of Large Groups.go
func largeGroupPositions(S string) [][]int {
    if len(S) < 3 {
        return [][]int{}
    }
    res := [][]int{}
    var end int
    for i := 0; i < len(S); i++ {
        end = i
        for end < len(S) - 1 && S[end] == S[end + 1] {
            end++
        }
        if (end - i + 1 >= 3) {
            res = append(res, []int{i, end})
        }
        i = end
    }
    return res
}
